pkg/virt-config/feature-gates.go

```diff
@@ -87,7 +86,6 @@ const (
 var deprecatedFeatureGates = [...]string{
 	LiveMigrationGate,
 	SRIOVLiveMigrationGate,
-	NonRoot,
```

Comment:
Some time ago a policy to deprecate a FG was created [1], which required old FGs to be left around, warning users that it is no longer in use.

Therefore, this is not exactly "dead code".
To remove this, we need to explain how it fits the policy or to update the policy.

Generally speaking, I am in favor to remove it, making sure the policy has such a path (e.g. after 2 versions, remove it completely but make sure it does not break old configurations).

[1] https://github.com/kubevirt/kubevirt/pull/8047

Reply:
I dropped the commit which removed the `NonRoot` feature gate.